synapse/replication/slave/storage/account_data.py

```diff
@@ -69,38 +69,25 @@ def stream_positions(self):
         result["tag_account_data"] = position
         return result
 
-    def process_replication(self, result):
-        stream = result.get("user_account_data")
-        if stream:
-            self._account_data_id_gen.advance(int(stream["position"]))
-            for row in stream["rows"]:
-                position, user_id, data_type = row[:3]
-                self.get_global_account_data_by_type_for_user.invalidate(
-                    (data_type, user_id,)
-                )
-                self.get_account_data_for_user.invalidate((user_id,))
+    def process_replication_rows(self, stream_name, token, rows):
+        if stream_name == "tag_account_data":
```

Comment:
This was done as part of making it such that you only get the stream you asked for, as all three would come down if you asked for "account_data". I appreciate it would have been better to do it separately